docs: clarifications to Sustainability Effort process (#12547)
argoproj/argo-workflows: community/sustainability_effort.md

### Authoring PRs

> Members of the repository have permission to apply labels to PRs and can therefore apply a `prioritized-review` label to any PRs they author.
> Those who are not yet members can request to have the label added by posting to the `#argo-wf-sustainability` channel.
> Feel free to give a gentle nudge in that channel if a PR you've authored has almost reached a week without an Assignee.

Reviewer: ```suggestion
Participants can apply a `prioritized-review` label to any PRs they author.
If you do not have permissions to add labels yet, you can request the label in the `#argo-wf-sustainability` channel.

If a PR has reached a week without an Assignee, you can request an assignment in the channel.
```

- simpler and more direct wording, [per k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
  - "Members" are of the org, not of the repository
  - We don't need to describe permissions and how they work here, the org-level roles document does that
    - We can just tell them what to do if they don't have that permission, which is substantially simpler and more direct.
    
 - add a new line for the "nudge" as that's a separate topic from labeling.
 - make the language for the "nudge" the same as the label for consistency and simplicity -- you can "request" both.
